src/backends/pyplot.jl

```diff
@@ -1139,6 +1139,11 @@ function _before_layout_calcs(plt::Plot{PyPlotBackend})
         # this sets the bg color inside the grid
         ax[set_facecolor_sym](py_color(sp[:background_color_inside]))
 
+        # link axes
+        if plt[:link] != :none
+            ax[:get_shared_x_axes]()[:join](ax, sp[:xaxis].sps[1].o)
+            ax[:get_shared_y_axes]()[:join](ax, sp[:yaxis].sps[1].o)
```

Comment:
why do you only check if `link != :none`? Other possibilities are `:x`, `:y`, `:both` and `:all`

Reply:
I don't want to link the axes when `link` isn't directly specified as the other possibilities you mentioned. Linking the axes to themselves caused issues with polar plots and I think heatmaps... I still need to check this out properly.